### DISCUSSION TITLE
Linter for checking span usage (End, SetStatus, RecordError)

### DISCUSSION THREAD

**Discussion by jjti** [open-telemetry/opentelemetry-go#4802] [Show and tell]
Hey! I'm sharing a linter I made for OpenTelemetry spans: https://github.com/jjti/go-spancheck

I've been converted to tracing, and use it religiously, but find it's easy to get spans "wrong":
- OOM-kills in prod because of missing `span.End()`s
- useless spans that were never updated with `span.SetStatus` `status:error` despite an error being returned
- missing `span.RecordError()`s that would have made debugging easier

This linter is an attempt to catch those common OTel span gotchus. Feedback welcome!

**Comment by MrAlias** [comment]
@jjti very cool :rocket: 

Have you thought about contributing this to `golangci-lint` as a linter?

**Reply by jjti** [reply]
Hey @MrAlias and thanks! I did add it to golangci-lint and it should be in the next release: https://github.com/golangci/golangci-lint/pull/4290. I'll update the docs as soon as it's in

**Reply by pellared** [reply]
Released https://github.com/golangci/golangci-lint/releases/tag/v1.56.0 🎉
